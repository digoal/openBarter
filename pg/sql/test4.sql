SET client_min_messages = warning;
\set ECHO none
\i uninstall_openbarter.sql
\i openbarter.sql
set search_path=market;
\set ECHO all
RESET client_min_messages;
select setval('ob.tdraft_id_seq',100);
select market.fcreate_quality('q1');
select market.fcreate_quality('q2');
select market.fcreate_quality('q3');
select market.fcreate_quality('q4');
select market.fcreate_quality('q5');
select market.fcreate_quality('q6');

select market.fadd_account('o1','olivier>q1',1000);
select market.fadd_account('o2','olivier>q2',1000);
select market.fadd_account('o3','olivier>q3',1000);

select market.finsert_bid('o1','olivier>q1',100,50,'olivier>q3');
select market.finsert_bid('o2','olivier>q2',100,50,'olivier>q1');
select market.finsert_bid('o3','olivier>q3',100,50,'olivier>q2');
-- executes getdraft_get(6,2,3,2)

-- should be 0
SELECT corrupted_stock_a+corrupted_stock_s+unbananced_qualities+corrupted_draft as errors from market.fstats();

select market.faccept_draft(100,'o1');
select market.faccept_draft(100,'o2');
select market.faccept_draft(100,'o3');

-- should be 0
SELECT corrupted_stock_a+corrupted_stock_s+unbananced_qualities+corrupted_draft as errors from market.fstats();

select qown,qname,owner,qtt from market.vowned;
select qown,qname,qtt from market.vbalance;
select id,did,provider,nat,qtt,receiver from market.vmvt;






