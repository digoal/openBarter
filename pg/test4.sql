/*truncate table ob_tlomega restart identity cascade;
truncate table ob_tomega restart identity cascade;

truncate table ob_tcommit restart identity ;
truncate table ob_tdraft restart identity cascade;

truncate table ob_tnoeud restart identity cascade;
truncate table ob_tstock restart identity cascade;
truncate table ob_towner restart identity cascade;
insert into ob_towner (name) values ('market');
truncate table ob_tquality restart identity cascade;
-- truncate table ob_tdepositary restart identity cascade;
truncate table ob_tmvt restart identity cascade;
truncate table ob_tldraft restart identity cascade;*/
SET client_min_messages = warning;
\set ECHO none
\i uninstall_openbarter.sql
\i openbarter.sql
\set ECHO all
RESET client_min_messages;
select setval('ob_tdraft_id_seq',100);

select ob_fcreate_quality('q1');
select ob_fcreate_quality('q2');
select ob_fcreate_quality('q3');
select ob_fcreate_quality('q4');
select ob_fcreate_quality('q5');
select ob_fcreate_quality('q6');

select ob_fadd_account('o1','olivier>q1',1000);
select ob_fadd_account('o2','olivier>q2',1000);
select ob_fadd_account('o3','olivier>q3',1000);

select ob_finsert_bid('o1','olivier>q1',100,50,'olivier>q3');
select ob_finsert_bid('o2','olivier>q2',100,50,'olivier>q1');
select ob_finsert_bid('o3','olivier>q3',100,50,'olivier>q2');






