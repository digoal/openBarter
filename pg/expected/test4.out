/*truncate table ob.tlomega restart identity cascade;
truncate table ob.tomega restart identity cascade;

truncate table ob.tcommit restart identity ;
truncate table ob.tdraft restart identity cascade;

truncate table ob.tnoeud restart identity cascade;
truncate table ob.tstock restart identity cascade;
truncate table ob.towner restart identity cascade;
insert into ob.towner (name) values ('market');
truncate table ob.tquality restart identity cascade;
-- truncate table ob.tdepositary restart identity cascade;
truncate table ob.tmvt restart identity cascade;
truncate table ob.tldraft restart identity cascade;*/
SET client_min_messages = warning;
\set ECHO none
 setval 
--------
      1
(1 row)

RESET client_min_messages;
select setval('ob.tdraft_id_seq',100);
 setval 
--------
    100
(1 row)

select ob.fcreate_quality('q1');
 fcreate_quality 
-----------------
               0
(1 row)

select ob.fcreate_quality('q2');
 fcreate_quality 
-----------------
               0
(1 row)

select ob.fcreate_quality('q3');
 fcreate_quality 
-----------------
               0
(1 row)

select ob.fcreate_quality('q4');
 fcreate_quality 
-----------------
               0
(1 row)

select ob.fcreate_quality('q5');
 fcreate_quality 
-----------------
               0
(1 row)

select ob.fcreate_quality('q6');
 fcreate_quality 
-----------------
               0
(1 row)

select ob.fadd_account('o1','olivier>q1',1000);
 fadd_account 
--------------
            0
(1 row)

select ob.fadd_account('o2','olivier>q2',1000);
 fadd_account 
--------------
            0
(1 row)

select ob.fadd_account('o3','olivier>q3',1000);
 fadd_account 
--------------
            0
(1 row)

select ob.finsert_bid('o1','olivier>q1',100,50,'olivier>q3');
 finsert_bid 
-------------
           0
(1 row)

select ob.finsert_bid('o2','olivier>q2',100,50,'olivier>q1');
 finsert_bid 
-------------
           0
(1 row)

select ob.finsert_bid('o3','olivier>q3',100,50,'olivier>q2');
 finsert_bid 
-------------
           1
(1 row)

-- executes ob.getdraft_get(6,2,3,2)
-- should be 0
SELECT corrupted_stock_a+corrupted_stock_s+unbananced_qualities+corrupted_draft as errors from ob.fstats();
 errors 
--------
      0
(1 row)

select ob.faccept_draft(100,'o1');
 faccept_draft 
---------------
             0
(1 row)

select ob.faccept_draft(100,'o2');
 faccept_draft 
---------------
             0
(1 row)

select ob.faccept_draft(100,'o3');
 faccept_draft 
---------------
             0
(1 row)

-- should be 0
SELECT corrupted_stock_a+corrupted_stock_s+unbananced_qualities+corrupted_draft as errors from ob.fstats();
 errors 
--------
      0
(1 row)

select qown,qname,owner,qtt from ob.vowned;
  qown   |   qname    | owner | qtt 
---------+------------+-------+-----
 olivier | olivier>q3 | o3    | 900
 olivier | olivier>q2 | o2    | 900
 olivier | olivier>q3 | o1    | 100
 olivier | olivier>q1 | o1    | 900
 olivier | olivier>q1 | o2    | 100
 olivier | olivier>q2 | o3    | 100
(6 rows)

select qown,qname,qtt from ob.vbalance;
  qown   |   qname    | qtt  
---------+------------+------
 olivier | olivier>q3 | 1000
 olivier | olivier>q2 | 1000
 olivier | olivier>q1 | 1000
(3 rows)

select id,did,provider,nat,qtt,receiver from ob.vmvt;
 id | did | provider |    nat     | qtt  | receiver 
----+-----+----------+------------+------+----------
  1 |     | market   | olivier>q1 | 1000 | o1
  2 |     | market   | olivier>q2 | 1000 | o2
  3 |     | market   | olivier>q3 | 1000 | o3
  4 |   4 | o1       | olivier>q1 |  100 | o2
  5 |   4 | o2       | olivier>q2 |  100 | o3
  6 |   4 | o3       | olivier>q3 |  100 | o1
(6 rows)

